### DISCUSSION TITLE
Force Pjax on Same Location

### DISCUSSION THREAD

**Discussion by anthonyfrancq** [PaperStrike/Pjax#102] [Ideas]
Hi there,

Is it possible to force pjax to load even when the link being clicked on is the current url? It appears that pjax is blocked when the link being clicked on is already the current url.

**Comment by PaperStrike** [comment]
Yes, links with the same path (pathname + query string) trigger no network request by design.

For 2.0.0+, you may listen to the anchor yourself, and package the link to a Request object. Pjax always sends Request objects without checking their paths.

```js
theLinkYouWant.addEventListener('click', (e) => {
  e.preventDefault();
  pjax.load(new Request(e.currentTarget.href));
});
```

If you are interested in native Pjax support, change the category of this discussion to **Ideas** (or open a new discussion there as you wish) and I will investigate later.

**Reply by anthonyfrancq** [reply]
Updated.

**Reply by PaperStrike** [reply]
@anthonyfrancq Sorry for the late response. I would like to implement this after I found not doing so is diverging from browsers' default behavior, but I forgot to reply or relabel this at that time. Thank you for your suggestion.
If you are wondering about the date, I'm not sure. I am trying to refactor the unit test system in this repo, and preparing for something important to me in the real life these days. I can only be sure that this won't happen before 12 September.

**Comment by PaperStrike** [comment]
Implemented in 2.1.0! Thank you!
